src/Languages/Properties/InterfaceProperties.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Metadata on a language's in depth interface properties.
+ */
+export class InterfaceProperties {
+
+    /**
+     * Keyword that indicates interface declaration.
+     */
+    public declareStartLeft: string;
+
+    /**
+     * Starts interface declaration.
+     */
+    public declareStartRight: string;
+
+    /**
+     * Keyword that indicates to extend from parent interfaces.
+     */
+    public declareExtendsLeft: string;
+
+    /**
+     * Seperator of multiple parent interfaces.
+     */
+    public declareExtendsRight: string;
+
+    /**
+     * Ends interface declaration.
+     */
+    public declareEnd: string;
+
+    /**
+     * Whether language supports interfaces.
+     */
+    public interfaceSupport: boolean;
```

Comment:
> `interfaceSupport`

Seems a little redundant to say "interface" in the name. How about just `supported`?